media-gfx/qimgv: version bump and remove old

kamiyaa opened the pull request:
media-gfx/qimgv: version bump 0.8.2
media-gfx/qimgv: remove old 0.7.3

juippis commented on media-gfx/qimgv/qimgv-0.8.2.ebuild:
```diff
@@ -0,0 +1,52 @@
+# Copyright 1999-2018 Gentoo Foundation
```
Old copyright header which makes me believe you didn't commit this with repoman like your git commit message makes it believe...

kamiyaa replied:
Oh, I didn't know you were supposed to commit with repoman. I didn't even know you could commit with repoman haha.
I just had it in my commit message because it was stated in the commit message formatting

juippis replied:
Using repoman to commit is optional, but couraged. The line in your commit message, `Package-Manager: Portage-2.3.69, Repoman-2.3.16` usually just implies it was made with repoman ;) you shouldn't write this automatically.

juippis commented on media-gfx/qimgv/qimgv-0.8.2.ebuild:
```diff
@@ -0,0 +1,52 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
```
7 should work with this.

juippis commented on media-gfx/qimgv/qimgv-0.8.2.ebuild:
```diff
@@ -0,0 +1,52 @@
+
+EAPI=6
+
+inherit cmake-utils gnome2-utils xdg-utils
```
Drop gnome2-utils and use `xdg_icon_cache_update` instead.

juippis commented on media-gfx/qimgv/qimgv-0.8.2.ebuild:
```diff
@@ -0,0 +1,52 @@
+
+inherit cmake-utils gnome2-utils xdg-utils
+
+DESCRIPTION="A cross-platform image viewer with webm support. Written in qt5"
```
`...webm support, written in qt5"`

juippis commented on media-gfx/qimgv/qimgv-0.8.2.ebuild:
```diff
@@ -0,0 +1,52 @@
+"
+DEPEND="
+	${RDEPEND}
+	virtual/pkgconfig
```
(With EAPI-7 this is a BDEPEND)

kamiyaa commented:
Resolved issues

juippis commented on media-gfx/qimgv/qimgv-0.8.2.ebuild:
```diff
@@ -0,0 +1,52 @@
+# Copyright 1999-2019 Gentoo Foundation
```
Still old header. New format is `Gentoo Authors` instead of `Foundation`.

juippis commented on media-gfx/qimgv/qimgv-0.8.2.ebuild:
```diff
@@ -0,0 +1,52 @@
+KEYWORDS="~amd64 ~x86"
+IUSE="kde video"
+
+BDEPEND="
```
This should be DEPEND instead.

juippis commented on media-gfx/qimgv/qimgv-0.8.2.ebuild:
```diff
@@ -0,0 +1,52 @@
+	kde? ( kde-frameworks/kwindowsystem:5 )
+	video? ( media-video/mpv[libmpv] )
+"
+DEPEND="
```
And this BDEPEND.

juippis commented on media-gfx/qimgv/qimgv-0.8.2.ebuild:
```diff
@@ -0,0 +1,52 @@
+	video? ( media-video/mpv[libmpv] )
+"
+DEPEND="
+	${BDEPEND}
```
BDEPEND is a split off from DEPEND, you shouldn't inherit either inside one another.
Also add RDEPEND somewhere in this ebuild, most likely inheriting the contents of DEPEND.

juippis commented on media-gfx/qimgv/qimgv-0.8.2.ebuild:
```diff
@@ -0,0 +1,52 @@
+}
+
+pkg_postrm() {
+	gnome2_icon_cache_update
```
Did you test building your ebuild after updating it? This should print out an error ;)
Update to xdg_icon_cache_update.

juippis commented on media-gfx/qimgv/qimgv-0.8.2.ebuild:
```diff
@@ -14,23 +14,25 @@ SLOT="0"
-DEPEND="
-	${RDEPEND}
-	virtual/pkgconfig
+RDEPEND="
```
`RDEPEND="${DEPEND}"`, I see you haven't really understood what BDEPEND is, maybe read some documentation on it? ;)

juippis commented on media-gfx/qimgv/qimgv-0.8.2.ebuild:
```diff
@@ -14,23 +14,25 @@ SLOT="0"
 	dev-qt/qtgui:5
 	dev-qt/qtwidgets:5
 	kde? ( kde-frameworks/kwindowsystem:5 )
 	video? ( media-video/mpv[libmpv] )
```
This seems like an unconditional dep, or it is controlled by some other flag.
```
[ebuild  N     ] media-gfx/qimgv-0.8.2::testworld  USE="-kde -video" 232 KiB

...

/tmp/portage/media-gfx/qimgv-0.8.2/work/qimgv-0.8.2/qimgv_libs/qimgv_player_mpv/src/mpvwidget.h:5:10: fatal error: mpv/client.h: No such file or directory
    5 | #include <mpv/client.h>
      |          ^~~~~~~~~~~~~~
compilation terminated.
```

kamiyaa commented:
- Removed `video` use flag as it is no longer used in build system
 - Added `exif` use flag

> BDEPEND is a split off from DEPEND, you shouldn't inherit either inside one another.
Also add RDEPEND somewhere in this ebuild, most likely inheriting the contents of DEPEND.

Sorry, I read this as inheriting the contents of BDEPEND xD

If I understand correctly:
`BDEPEND`: is for the stuff needed to compile it (pkgconfig, cmake)
`DEPEND`: is for the stuff needed by the applicationt to run (qt5).

Then what is the new usage of `RDEPEND`? Backwards compatibility?

juippis commented:
BDEPEND is a split from DEPEND. Basically, BDEPEND = binaries just needed to execute during build-time, while DEPEND = libraries and headers. It's got to do with cross-compile, not everything need to be compiled on the computer the final product gets installed in.
There's more depth to all this, but that's the simple explanation.
